packages/snaps-cli/src/webpack/loaders/wasm.ts

```diff
@@ -1,27 +1,31 @@
-import { assert, stringToBytes } from '@metamask/utils';
+import { assert } from '@metamask/utils';
 
 /**
- * A Webpack loader that inlines the WASM module as a `Uint8Array`. This makes
- * it possible to import the WASM module directly, and use it with the
- * `WebAssembly.instantiate` function.
- *
- * This is useful, because snaps are not allowed to import assets from outside
- * of their package. This loader allows you to inline the WASM module as a
- * `Uint8Array`, which can then be passed to `WebAssembly.instantiate`.
+ * A Webpack loader that synchronously loads the WASM module. This makes it
+ * possible to import the WASM module directly.
  *
  * @param source - The WASM module as a string.
  * @returns A string that exports the WASM module as a `Uint8Array`.
  * @example
  * ```ts
  * import wasm from './program.wasm';
  *
- * const { instance } = await WebAssembly.instantiate(wasm, {});
  * // Do something with the WASM module...
  * ```
  */
 export default function loader(source: unknown) {
-  assert(typeof source === 'string', 'Expected source to be a string.');
+  assert(source instanceof Uint8Array, 'Expected source to be a Uint8Array.');
+
+  return `
+    const bytes = new Uint8Array(${JSON.stringify(Array.from(source))});
+    const module = new WebAssembly.Module(bytes);
+    const instance = new WebAssembly.Instance(module, {});
```

Comment:
Hmm with this implementation it becomes harder to pass imports to the WASM instance. Wondering if that will be a problem down the line. But potentially a reason for us to support multiple ways of doing this, allowing for inlining of files as an `Uint8Array` as well

Reply:
I've added support for imports.